Utilities/DavixAdaptor/src/DavixFile.cc

```diff
@@ -0,0 +1,363 @@
+                              DavixError **davixErr) {
+  std::string ucert, ukey;
+  char default_proxy[64];
+  snprintf(default_proxy, sizeof(default_proxy), "/tmp/x509up_u%d", geteuid());
+  // X509_USER_PROXY
+  if (getenv("X509_USER_PROXY")) {
+    edm::LogInfo("DavixFile") << "X509_USER_PROXY found in environment."
+                              << " Will use it for authentication";
+    ucert = ukey = getenv("X509_USER_PROXY");
+  }
+  // Default proxy location
+  else if (access(default_proxy, R_OK) == 0) {
+    edm::LogInfo("DavixFile") << "Found proxy in default location " << default_proxy
+                              << " Will use it for authentication";
+    ucert = ukey = default_proxy;
+  }
+  // X509_USER_CERT
+  else if (getenv("X509_USER_CERT")) {
+    ucert = getenv("X509_USER_CERT");
+  }
+  // X509_USER_KEY only if X509_USER_CERT was found
+  if (!ucert.empty() && getenv("X509_USER_KEY")) {
+    edm::LogInfo("DavixFile") << "X509_USER_{CERT|KEY} found in environment"
+                              << " Will use it for authentication";
+    ukey = getenv("X509_USER_KEY");
+  }
+  // Check if vars are set...
+  if (ucert.empty() || ukey.empty()) {
+    edm::LogWarning("DavixFile") << "Was not able to find proxy in $X509_USER_PROXY, "
+                                 << "X509_USER_{CERT|KEY} or default proxy creation location. "
+                                 << "Will try without authentication";
+    return -1;
+  }
+  return cert->loadFromFilePEM(ukey, ucert, "", davixErr);
+}
+
+void DavixFile::create(const char *name, bool exclusive /* = false */, int perms /* = 066 */) {
+  open(name, (IOFlags::OpenCreate | IOFlags::OpenWrite | IOFlags::OpenTruncate |
+              (exclusive ? IOFlags::OpenExclusive : 0)),
+       perms);
+}
+
+void DavixFile::create(const std::string &name, bool exclusive /* = false */,
+                       int perms /* = 066 */) {
+  open(name.c_str(), (IOFlags::OpenCreate | IOFlags::OpenWrite | IOFlags::OpenTruncate |
+                      (exclusive ? IOFlags::OpenExclusive : 0)),
+       perms);
+}
+
+void DavixFile::open(const std::string &name, int flags /* = IOFlags::OpenRead */,
+                     int perms /* = 066 */) {
+  open(name.c_str(), flags, perms);
+}
+
+void DavixFile::open(const char *name, int flags /* = IOFlags::OpenRead */, int perms /* = 066 */) {
+  // Actual open
+  if ((name == 0) || (*name == 0)) {
+    edm::Exception ex(edm::errors::FileOpenError);
+    ex << "Cannot open a file without name";
+    ex.addContext("Calling DavixFile::open()");
+    throw ex;
+  }
+
+  if ((flags & IOFlags::OpenRead) == 0) {
+    edm::Exception ex(edm::errors::FileOpenError);
+    ex << "Must open file '" << name << "' at least for read";
+    ex.addContext("Calling DavixFile::open()");
+    throw ex;
+  }
+
+  if (davixPosix && m_fd) {
+    edm::Exception ex(edm::errors::FileOpenError);
+    ex << "Davix::open(name='" << m_name->c_str() << "') failed on already open file";
```

Comment:
I suggest making m_name just a `std::string` since it will make memory management easier. I'd also set `m_name = name` at line 152 right after you've tested that `name` is valid.